src/kernels/raw/finder/localKnownPathKernelSpecFinder.node.ts

```diff
@@ -4,97 +4,133 @@
      */
     @capturePerfTelemetry(Telemetry.KernelListingPerf, { kind: 'localKernelSpec' })
-    public async listKernelSpecs(
-        includePythonKernels: boolean,
-        cancelToken?: CancellationToken
-    ): Promise<(LocalKernelSpecConnectionMetadata | PythonKernelConnectionMetadata)[]> {
-        return this.listKernelsWithCache(
-            includePythonKernels ? 'IncludePythonV2' : 'ExcludePythonV2',
-            false,
-            async () => {
-                // First find the on disk kernel specs and interpreters
-                const kernelSpecs = await this.findKernelSpecs(cancelToken);
+    private async listKernelSpecs(cancelToken: CancellationToken): Promise<LocalKernelSpecConnectionMetadata[]> {
+        return this.listKernelsWithCache('LocalKnownPathKernelSpecFinder', false, async () => {
+            // First find the on disk kernel specs and interpreters
+            const kernelSpecs = await this.findKernelSpecs(cancelToken);
 
-                return kernelSpecs
-                    .filter((item) => {
-                        if (includePythonKernels) {
-                            return true;
-                        }
-                        return item.language !== PYTHON_LANGUAGE;
-                    })
-                    .map(
-                        (k) =>
-                            <LocalKernelSpecConnectionMetadata>{
-                                kind: 'startUsingLocalKernelSpec',
-                                kernelSpec: k,
-                                interpreter: undefined,
-                                id: getKernelId(k)
-                            }
-                    );
+            const mappedKernelSpecs = kernelSpecs.map(
+                (k) =>
+                    <LocalKernelSpecConnectionMetadata>{
+                        kind: 'startUsingLocalKernelSpec',
+                        kernelSpec: k,
+                        interpreter: undefined,
+                        id: getKernelId(k)
+                    }
+            );
+            if (cancelToken.isCancellationRequested) {
+                return [];
             }
-        );
+            const oldKernels = this._cachedKernels;
+            this._cachedKernels = mappedKernelSpecs;
+
+            // Trigger a change event if we have different kernels.
+            oldKernels.sort();
+            mappedKernelSpecs.sort();
+            if (
+                oldKernels.length !== mappedKernelSpecs.length ||
+                JSON.stringify(oldKernels) !== JSON.stringify(mappedKernelSpecs)
+            ) {
+                this._onDidChangeKernels.fire();
+            }
+            this._onDidChangeKernels.fire();
+            return mappedKernelSpecs;
+        });
+    }
+    private async initialize(): Promise<void> {
+        if (this._initialized) {
+            return this._initialized;
+        }
+        const cancellation = new CancellationTokenSource();
```

Comment:
Is the cancellation token here just a no-op in this case?

Reply:
Not following you, the calling methods needs it, hence we're passing one.
I didn't want to make it an optional argument as that complicates the code even more.
This block of code is only ever called once.

Reply:
I guess my question was why pass a cancellation token that does nothing. I assume something further down the chain requires it in some calls and not others?